drivers/ethernet/eth_stm32_hal_common.c

```diff
@@ -431,6 +431,18 @@ static const struct eth_stm32_hal_dev_cfg eth0_config = {
 #if DT_INST_CLOCKS_HAS_NAME(0, mac_clk_ptp)
 	.pclken_ptp = {.bus = DT_INST_CLOCKS_CELL_BY_NAME(0, mac_clk_ptp, bus),
 		       .enr = DT_INST_CLOCKS_CELL_BY_NAME(0, mac_clk_ptp, bits)},
+#endif
+#if DT_INST_CLOCKS_HAS_NAME(0, mac_clk)
+	.pclken_mac = {.bus = DT_INST_CLOCKS_CELL_BY_NAME(0, mac_clk, bus),
+		       .enr = DT_INST_CLOCKS_CELL_BY_NAME(0, mac_clk, bits)},
+#endif
+#if DT_INST_CLOCKS_HAS_NAME(0, eth_ker)
+	.pclken_ker = {.bus = DT_INST_CLOCKS_CELL_BY_NAME(0, eth_ker, bus),
+		       .enr = DT_INST_CLOCKS_CELL_BY_NAME(0, eth_ker, bits)},
+#endif
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32mp13_ethernet)
+	.clockselection = DT_NODE_HAS_PROP(0, st_ext_phyclk) ? HAL_ETH1_REF_CLK_RCC
+							     : HAL_ETH1_REF_CLK_RX_CLK_PIN,
```

Comment:
this is later copied over with
```
heth->Init.ClockSelection = cfg->clockselection;
```
why don't you set it a few lines down directly in `static struct eth_stm32_hal_dev_data eth0_data` after `.MediaInterface = STM32_ETH_PHY_MODE(0),`

Reply:
Yes that seems more optimized i removed `clockselection`  from the `eth_stm32_hal_dev_cfg` struct, thanks!